webapp/views/wpt-results.js

```diff
@@ -897,6 +929,92 @@ class WPTResults extends AmendMetadataMixin(Pluralizer(WPTColors(WPTFlags(PathIn
     this.path = this.searchResults[(n + next) % n].test;
   }
 
+  sortTestName() {
+    if (!this.displayedNodes) {
+      return;
+    }
+
+    this.isPathSorted = !this.isPathSorted;
+    this.sortRow = new Array(this.testRuns.length).fill(false);
+    const sortedNodes = this.displayedNodes.slice();
+    const self = this;
+    sortedNodes.sort(function (a, b) {
+      if (self.isPathSorted) {
+        return self.compareTestNameDefaultOrder(a, b);
+      }
+      return self.compareTestNameDefaultOrder(b, a);
+    });
+    this.displayedNodes = sortedNodes;
+  }
+
+  compareTestName(a, b) {
+    if (this.isPathSorted) {
+      return this.compareTestNameDefaultOrder(a, b);
+    }
+    return this.compareTestNameDefaultOrder(b, a);
+  }
+
+  compareTestNameDefaultOrder(a, b) {
+    const pathA = a.path.toLowerCase();
+    const pathB = b.path.toLowerCase();
+    if (pathA < pathB) {
+      return -1;
+    }
+
+    if (pathA > pathB) {
+      return 1;
+    }
+    return 0;
+  }
+
+  sortTestResults() {
+    const [index] = arguments;
+    return () => {
+      if (!this.displayedNodes) {
+        return;
+      }
+
+      const sortedNodes = this.displayedNodes.slice();
+      const self = this;
+      sortedNodes.sort(function (a, b) {
+        if (self.sortRow[index]) {
```

Comment:
I think `sortRow` could be represented by an integer here, being the index that is currently sorted on and compare it with the `index` var. The issue would be trying to switch to descending and ascending again when clicking the same column multiple times, but that should still work if you set this var to -1 whenever it was previously a match with the current index. It should remove the need to rebuild an array each time

Reply:
I agree but `sortRow` array is a cleaner way to reset all other arrows. Every time sortRow is being reassigned, it triggers (through data binding) `<template is="dom-repeat" items="[[sortRow]]" as="sortItem">` to repopulate the arrow icons. I didn't see a good way to do that if `sortRow` is a number